types/preview/@ember/template-compilation/index.d.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * References:
+ *  - https://github.com/emberjs/rfcs/pull/496
+ *  - https://github.com/emberjs/rfcs/pull/779
+ */
+declare module '@ember/template-compilation' {
+  interface TemplateFactory {
+    __htmlbars_inline_precompile_template_factory: any;
+  }
```

Comment:
We also have this branded interface defined here: https://github.com/emberjs/ember.js/blob/0c5518ea7b8eb6503e0d3d8a47153e0d619185be/types/preview/%40ember/component/index.d.ts#L11-L13

and here: https://github.com/emberjs/ember.js/blob/0c5518ea7b8eb6503e0d3d8a47153e0d619185be/types/preview/ember/index.d.ts#L374-L379

Seems like we should seek to have only one definition of this?

I notice we dropped the export when brining in [`htmlbars-inline-precompile` types from DT](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d38588fa1cd9fcab721c3ce78a24234b72d14ebc/types/htmlbars-inline-precompile/index.d.ts#L34-L36) (was never exported from [@ember/component](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d38588fa1cd9fcab721c3ce78a24234b72d14ebc/types/ember__component/index.d.ts#L19-L21)). I don't know that it necessarily makes sense to export from here, but this would make the third definition in these types.

If we _do_ export, I like this suggestion: 
> In the future, we will coordinate a change which makes this export only the type side of a class with a private property, such that it *cannot* be used.

from: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d38588fa1cd9fcab721c3ce78a24234b72d14ebc/types/htmlbars-inline-precompile/index.d.ts#L31-L33

@chriskrycho, @dfreeman thoughts?

Reply:
Maybe we should put it in https://github.com/emberjs/ember.js/blob/0c5518ea7b8eb6503e0d3d8a47153e0d619185be/types/preview/%40ember/component/-private/glimmer-interfaces.d.ts#L1

https://github.com/glimmerjs/glimmer-vm/blob/82ea580f8f4dcabfab98abc353b3a01891934f67/packages/%40glimmer/interfaces/lib/template.d.ts#L57

Reply:
Centralizing to a single definition seems good; I like that suggestion, @jamescdavis!

Reply:
It this PR or separate?

I'd prefer separate so PRs can remain focused

Reply:
Separate seems good.

Reply:
Ah, no, actually. 😂 Let's not *introduce another copy* here. Pull it from `ember.js/types/preview/@ember/component/-private/glimmer-interfaces.d.ts`, as @jamescdavis suggested. That isn't in conflict with PRs remaining focused—quite the contrary!

Reply:
done and pushed!